### PR TITLE
[DOM-56006] Regenerated the clients

### PR DESCRIPTION
**Pull request by ddl-joyce-zhao**
## Description

Regenerated the clients with the new version of openapi-python-client. Here is its release history:

https://github.com/openapi-generators/openapi-python-client/releases

To have the compatible pydantic, the openapi-python-client has to be equal to or greater than 0.15.1 where the pydantic was upgraded to V2. 
https://github.com/openapi-generators/openapi-python-client/releases/tag/v0.15.1

Noticeable breaking changes that affects our generated clients include 
https://github.com/openapi-generators/openapi-python-client/releases/tag/v0.15.0 

Useful discussion about the new type of clients:
https://github.com/openapi-generators/openapi-python-client/pull/775#issuecomment-1652252833

## Related Issue

https://dominodatalab.atlassian.net/browse/DOM-56006

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/dominodatalab/domino-data/blob/main/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: datasource_api_client/models/datasource_dto_auth_type.py [hunk 1]
```diff
@@ -2,16 +2,16 @@
 
 
 class DatasourceDtoAuthType(str, Enum):
-    AZUREBASIC = "AzureBasic"
-    BASIC = "Basic"
     AWSIAMBASIC = "AWSIAMBasic"
     AWSIAMBASICNOOVERRIDE = "AWSIAMBasicNoOverride"
     AWSIAMROLE = "AWSIAMRole"
     AWSIAMROLEWITHUSERNAME = "AWSIAMRoleWithUsername"
+    AZUREBASIC = "AzureBasic"
```
Comment (ddl-eric-jin) [inline datasource_api_client/models/datasource_dto_auth_type.py:9]: Nice alphabetization!

File: datasource_api_client/models/list_request.py [hunk 2]
```diff
@@ -58,6 +64,8 @@ def to_dict(self) -> Dict[str, Any]:
 
     @classmethod
     def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
+        from ..models.datasource_config import DatasourceConfig
```
Comment (ddl-eric-jin) [inline datasource_api_client/models/list_request.py:67]: I know this is auto-generated but just curious, why would they add the import inside the method?

File: feature_store_api_client/api/default/get.py [hunk 3]
```diff
@@ -1,86 +1,93 @@
...
 from ...models.feature_store import FeatureStore
 from ...types import Response
 
 
-def _get_kwargs(
-    *,
-    client: Client,
-) -> Dict[str, Any]:
-    url = "{}/".format(client.base_url)
+def _get_kwargs() -> Dict[str, Any]:
```
Comment (ddl-eric-jin) [inline feature_store_api_client/api/default/get.py:13]: Will not returning things like the headers, cookies, etc. like before cause any issues?

File: datasource_api_client/models/datasource_dto.py [hunk 4]
```diff
@@ -29,23 +33,23 @@ class DatasourceDto:
...
         status (DatasourceDtoStatus):
-        description (Union[Unset, None, str]):
+        description (Union[None, Unset, str]):
     """
 
-    added_by: DatasourceDtoAddedBy
+    added_by: "DatasourceDtoAddedBy"
     auth_type: DatasourceDtoAuthType
-    config: DatasourceConfig
+    config: "DatasourceConfig"
```
Comment (ddl-eric-jin) [inline datasource_api_client/models/datasource_dto.py:41]: Why is there a mix of Strings and variables for the type definitions?

**Review by ddl-eric-jin — APPROVED** [review]
Given that most of the changes are from auto-generated code, it looks good to me! I had a few comments/questions about the changes. It also would be good to run system tests with this client if you haven't already.

File: feature_store_api_client/api/default/get.py (hunk 3 above)
Comment (ddl-joyce-zhao) [reply feature_store_api_client/api/default/get.py:13]: Nice catch! I regenerated the client again and seems this is brought back.

File: datasource_api_client/models/list_request.py (hunk 2 above)
Comment (ddl-joyce-zhao) [reply datasource_api_client/models/list_request.py:67]: This is under the class method. And the data type is only used for converting from dictionary. I think the rational here is for lazy imports.

File: datasource_api_client/models/datasource_dto.py (hunk 4 above)
Comment (ddl-joyce-zhao) [reply datasource_api_client/models/datasource_dto.py:41]: Very good point. I haven't found out why it's doing so. But seeing it's widely used in open api python generator and generated code. https://github.com/openapi-generators/openapi-python-client/blob/main/openapi_python_client/parser/openapi.py#L192

**Pull request merged**